pilot/pkg/networking/core/listener.go

```diff
@@ -1111,18 +1114,20 @@ func buildGatewayListener(opts gatewayListenerOpts, transport istionetworking.Tr
 		filterChains = append(filterChains, &listener.FilterChain{
 			FilterChainMatch: match,
 			TransportSocket:  transportSocket,
+			// Setting this timeout enables the gateway to enhance its resistance against memory exhaustion attacks,
+			// such as slow TLS Handshake attacks.
+			TransportSocketConnectTimeout: durationpb.New(defaultGatewayTransportSocketConnectTimeout),
 		})
 	}
 
 	res := &listener.Listener{
 		TrafficDirection: core.TrafficDirection_OUTBOUND,
 		ListenerFilters:  listenerFilters,
 		FilterChains:     filterChains,
-		// For Gateways, we want no timeout. We should wait indefinitely for the TLS if we are sniffing.
-		// The timeout is useful for sidecars, where we may operate on server first traffic; for gateways if we have listener filters
-		// we know those filters are required.
-		ContinueOnListenerFiltersTimeout: false,
-		ListenerFiltersTimeout:           durationpb.New(0),
```

Comment:
I don't love that we're taking away functionality here; this field is no longer overridable. Should the proxy detection timeout override the default? Should we just add a new field?

Reply:
I feel that too many configuration options can confuse users, and if users really have the need to make changes, they can do so through EnvoyFilter. Here, the behavior remains consistent with before, using default values, but in a more straightforward manner, directly employing Envoy's default settings.

Reply:
FWIW I think this is effectively the same as the default in 1.19 that was unintentionally (I think?) change. I don't really see too much usage for someone to change the timeout here. Did you expect users would want to make it longer or shorter?

Reply:
Oh my bad, I thought the protocol detect timeout previously affected this. If this is the same as 1.19, then this LGTM

Reply:
#47970 intended to set no timeout

Reply:
> #47970 intended to set no timeout

ah thanks, forgot about that. 

I definitely think part of that PR is right - we should never continue on timeout. Whether we have no timeouts at all is another thing... it does seem like a good practice in general to have timeouts for everything especially at edge?